packages/keyring-api/src/api/transaction.ts

```diff
@@ -0,0 +1,296 @@
+ *   type: 'send',
+ *   from: [
+ *     {
+ *       address: 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ *   to: [
+ *     {
+ *       address: 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *     {
+ *       address: 'bc1qwl8399fz829uqvqly9tcatgrgtwp3udnhxfq4k',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ *   fees: [
+ *     {
+ *       type: 'priority',
+ *       asset: {
+ *         fungible: true,
+ *         type: 'bip122:000000000019d6689c085ae165831e93/slip44:0',
+ *         unit: 'BTC',
+ *         amount: '0.1',
+ *       },
+ *     },
+ *   ],
+ * };
+ * ```
+ */
+export const TransactionStruct = object({
+  /**
+   * Chain-specific transaction ID.
+   */
+  id: string(),
+
+  /**
+   * Chain ID (CAIP-2).
+   */
+  chain: CaipChainIdStruct,
+
+  /**
+   * Account ID (UUIDv4).
+   */
+  account: UuidStruct,
+
+  /**
+   * Transaction status {@see TransactionStatus}.
+   */
+  status: enums([
+    `${TransactionStatus.Submitted}`,
+    `${TransactionStatus.Unconfirmed}`,
+    `${TransactionStatus.Confirmed}`,
+    `${TransactionStatus.Failed}`,
+  ]),
+
+  /**
+   * Transaction type. This will be used by MetaMask to enrich the transaction
+   * details on the UI.
+   *
+   * The possible values are:
+   *
+   * - send: The transaction was originated by the account. If the transaction
+   * has a change output that goes back to the same account, it must be tagged
+   * as a send transaction.
+   *
+   * - receive: The transaction was received by the account, but originated by
+   * another account.
+   */
+  type: enums([`${TransactionType.Send}`, `${TransactionType.Receive}`]),
+
+  /**
+   * Transaction sender addresses and amounts.
+   */
+  from: array(ParticipantStruct),
+
+  /**
+   * Transaction receiver addresses and amounts.
+   */
+  to: array(ParticipantStruct),
+
+  /**
+   * Total transaction fee.
+   */
+  fees: array(FeeStruct),
+
+  /**
+   * List of events related to the transaction.
+   *
+   * The events are tracked in a best-effort basis and may not be available for
+   * all transactions.
+   */
+  events: array(
+    object({
+      /**
+       * New status of the transaction.
+       */
+      status: enums([
+        `${TransactionStatus.Submitted}`,
+        `${TransactionStatus.Unconfirmed}`,
+        `${TransactionStatus.Confirmed}`,
+        `${TransactionStatus.Failed}`,
+      ]),
+
+      /**
+       * UNIX timestamp of when the event was occurred.
```

Comment:
Not sure about that one?

```suggestion
       * UNIX timestamp of when the event occurred.
```